tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -405,20 +414,29 @@ def get_inputs_to_model(self, model, X, model_pred_proba_dict=None, fit=False, p
             X = model.preprocess(X=X, preprocess_stateful=False)
         return X
 
-    def score(self, X, y, model=None) -> float:
+    def score(self, X, y, model=None, weights=None) -> float:
         if self.eval_metric.needs_pred:
-            y_pred_ensemble = self.predict(X=X, model=model)
-            return self.eval_metric(y, y_pred_ensemble)
+            y_pred = self.predict(X=X, model=model)
         else:
-            y_pred_proba_ensemble = self.predict_proba(X=X, model=model)
-            return self.eval_metric(y, y_pred_proba_ensemble)
+            y_pred = self.predict_proba(X=X, model=model)
+        if not self.weight_evaluation:
```

Comment:
same comment as earlier, these 5 lines can be simplified to 1 by expanding `compute_weighted_metric` functionality.